uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -219,9 +245,10 @@ impl KotlinCodeOracle {
                 Box::new(compounds::MapCodeType::new(inner, outer))
             }
             Type::External { .. } => panic!("no support for external types yet"),
-            Type::Wrapped { name, prim } => Box::new(wrapped::WrappedCodeType::new(
-                name,
-                self.create_code_type(prim.as_ref().clone()),
+            Type::Custom { name, builtin } => Box::new(custom::CustomCodeType::new(
+                name.clone(),
+                builtin.as_ref().clone(),
+                self.config.custom_types.get(&name).cloned(),
```

Comment:
I wonder if you can simplify things here by checking 

```rust
if Some(config) = self.config.custom_types.get(&name) {
    Box::new(custom::CustomCodeType::new(name.clone(), builtin.as_ref().clone(), config.clone())
} else {
    self.create_code_type(builtin.as_ref().clone())
}
```

and only invoke `CustomType` templates and `CodeType` and `CodeDecalration` when the config exists.

Reply:
This gets into a requirement that I've been half-committed to: having a consistent FFI conversion API for all types.  The end goal there would be that you can support external types by importing a `FFIConverter` subclass and calling it's methods, without knowing much about the underlying type.  I say "half-committed" because I'm not really sure how it's all going to turn out, but I'm going to try at least.

Reply:
I had a little go, and it seemed like a string that we could pull on in a followup. 

I think a unified `FFIConverter<T, U>` interface is a good goal too, but I think that's orthogonal to this.